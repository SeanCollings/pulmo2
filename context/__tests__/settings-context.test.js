import React, { useContext } from 'react';
import { create } from 'react-test-renderer';
import { cleanup } from 'react-native-testing-library';
import SettingsContextProvider, { SettingsContext } from '../settings-context';

jest.mock('../../helpers/storage');

const MockComponent = () => {
  const context = useContext(SettingsContext);
  expect(context).toMatchSnapshot();
  return null;
};

describe('settings-context', () => {
  afterEach(cleanup);

  test('unit test', () => {
    let context = create(
      <SettingsContextProvider>
        <MockComponent />
      </SettingsContextProvider>
    );

    expect(context).toMatchSnapshot();
  });
});
