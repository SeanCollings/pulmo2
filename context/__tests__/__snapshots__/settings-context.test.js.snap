// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings-context unit test 1`] = `
Object {
  "settingsContext": Object {
    "theme": "light_mode",
  },
  "updateSettingsContext": [Function],
}
`;

exports[`settings-context unit test 2`] = `null`;
